llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -158,11 +158,16 @@ class MemsetRanges {
 
   /// A sorted list of the memset ranges.
   SmallVector<MemsetRange, 8> Ranges;
-
-  const DataLayout &DL;
+  const DataLayout *DL;
 
 public:
-  MemsetRanges(const DataLayout &DL) : DL(DL) {}
+  MemsetRanges() {}
+  MemsetRanges(const DataLayout *DL) : DL(DL) {}
```

Comment:
Where is the DL-only overload used?

Reply:
No, I keep it just because its original implementation contains it...